### PR TITLE
emails works with smtp debugserver, still needs to test real ones.

### PR DESCRIPTION
**Pull request by ystaggl**
Also added example dependency (python3)

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -144,10 +147,40 @@ def check_date(store, date_to_check):
...
         if op != 'equal'
-    ]
+    ] #Contains the difference between the current details and the previous history file.
 
     if diff:
         pprint(diff)
+        send_email(diff)
+
+def send_email(diff):
+    email_body = str(diff).replace('–','-').replace('\', \'', '\',\n\'') #Changing the dashes is more legible from terminal, can remove that for real emails.
```
Comment (robinchew) [inline main.py]: make a change to the diff itself rather than replace if you really want to change the format

File: main.py [hunk 2]
```diff
@@ -144,10 +147,40 @@ def check_date(store, date_to_check):
...
+    sent_from = 'sender@obsi.com.au'
+    to = ['robinchew@gmail.com']
+    gmail_user='mb36340'
```
Comment (robinchew) [inline main.py]: environment variable for from, to and user

File: main.py [hunk 3]
```diff
@@ -144,10 +147,40 @@ def check_date(store, date_to_check):
...
+    gmail_password = ''
+
+    subject = 'New booking details'
+    email_text = f"""\
+    Message-ID: <{uuid4()}@obsi.com.au>
```
Comment (robinchew) [inline main.py]: environment variable for the domain part 'obsi.com.au'. Can be extracted from the FROM environment variable

File: main.py [hunk 4]
```diff
@@ -144,10 +147,40 @@ def check_date(store, date_to_check):
...
+    From: Python SMTP Sender <%s>
+    To: %s
+    Subject: %s
+
+    %s
+    """ % (sent_from, ", ".join(to), subject, email_body)
+    try:
+        smtp_server = smtplib.SMTP_SSL('relay.mailbaby.net', 465)
```
Comment (robinchew) [inline main.py]: environment variable for server host and port. If port is 465, use ssl function

File: main.py [hunk 5]
```diff
@@ -144,10 +147,40 @@ def check_date(store, date_to_check):
...
         for op, astart, aend, bstart, bend in  difflib.SequenceMatcher(None, last, booked_list).get_opcodes()
         if op != 'equal'
-    ]
+    ] #Contains the difference between the current details and the previous history file.
 
     if diff:
         pprint(diff)
+        send_email(diff)
+
+def send_email(diff):
```
Comment (robinchew) [inline main.py]: with all the environment variable comments below, they should come from send_email arguments:

```python
def send(email_host, email_port, user, password, from, to):
```

Then when calling send, fill the arguments from environment variables. 

Passing arguments is better than accessing os.environ which is essentially a global. Only use os.environ contained in one spot, then pass around as arguments

Comment (robinchew) [reply main.py]: Also with these arguments send_email becomes more useful and able to be used on multiple servers and send to different recipients, amongst other things.

File: main.py [hunk 6]
```diff
@@ -144,10 +147,44 @@ def check_date(store, date_to_check):
...
+                   os.environ['EMAIL_PORT'],
+                   os.environ['EMAIL_USER'],
+                   os.environ['EMAIL_PASSWORD'],
+                   os.environ['EMAIL_FROM'],
+                   os.environ['EMAIL_TO'],
+                   str(diff).encode('utf-8'), #Encoding is necessary due to the en-dash character.
+                   'Updated booking details')
+
+def send_email(email_host, email_port, user, password, email_from, email_to, email_body, email_subject):
+    email_domain = email_from.split('@')[1]
```
Comment (robinchew) [inline main.py]: Just a minor improvement:
```python
_user, email_domain = email_from.split('@', 1) # 1 means split on first occurence of '@'
```

File: build_badtrack.py [hunk 7]
```diff
@@ -30,6 +30,7 @@
 control_content = """\
 Package: badtrack
 Version: 1.0.0
+Depends: python3
```
Comment (robinchew) [inline build_badtrack.py:33]: How do you define multiple dependencies?

File: main.py [hunk 8]
```diff
@@ -144,10 +147,44 @@ def check_date(store, date_to_check):
         '{}: {} -> {}'.format(op, last[astart:aend], booked_list[bstart:bend])
         for op, astart, aend, bstart, bend in  difflib.SequenceMatcher(None, last, booked_list).get_opcodes()
         if op != 'equal'
-    ]
+    ] #Contains the difference between the current details and the previous history file.
 
     if diff:
         pprint(diff)
+        send_email(os.environ['EMAIL_HOST'],
+                   os.environ['EMAIL_PORT'],
```
Comment (robinchew) [inline main.py]: Gotta do `int(os.environ['EMAIL_PORT'])`

File: build_badtrack.py (hunk 7 above)
Comment (ystaggl) [reply build_badtrack.py:33]: https://www.debian.org/doc/debian-policy/ch-relationships.html (it's just comma separated)
I'll just add it as a comment so we don't need to look for it later.

File: main.py [hunk 9]
```diff
@@ -144,10 +147,44 @@ def check_date(store, date_to_check):
...
 
     if diff:
         pprint(diff)
+        send_email(os.environ['EMAIL_HOST'],
+                   int(os.environ['EMAIL_PORT']),
+                   os.environ['EMAIL_USER'],
+                   os.environ['EMAIL_PASSWORD'],
+                   os.environ['EMAIL_FROM'],
+                   os.environ['EMAIL_TO'],
+                   str(diff).encode('utf-8'), #Encoding is necessary due to the en-dash character.
```
Comment (robinchew) [inline main.py]: Isn't this enough?
```python
diff.encode('utf-8')
```

Comment (ystaggl) [reply main.py]: No, it causes an error because diff is a list. I actually tried that first.

Comment (robinchew) [reply main.py]: in that case:
```python
pprint.pformat(diff).encode('utf-8')
```

**Pull request merged**